### PR TITLE
:ambulance: Add https/http for InfluxDB fixes #25

### LINKED ISSUE #25: Glances no longer writes to influxDB
# Problem/Motivation

> I noticed after updating hassio and all add-ons that glances is no longer writing to InfluxDB.

## Expected behavior

> Data continuing to be written to the database

## Actual behavior

> On startup of the Glances add-on in hassio, I get the following alert:
INFO: Starting Glances...
INFO: Starting NGinx...
nginx: [alert] detected a LuaJIT version which is not OpenResty's; many optimizations will be disabled and performance will be compromised (see https://github.com/openresty/luajit2 for OpenResty's LuaJIT or, even better, consider using the OpenResty releases from https://openresty.org/en/download.html)
Traceback (most recent call last):
  File "/usr/bin/glances", line 11, in <module>
    load_entry_point('Glances==3.1.1', 'console_scripts', 'glances')()
  File "/usr/lib/python3.7/site-packages/glances/__init__.py", line 140, in main
    start(config=config, args=args)
  File "/usr/lib/python3.7/site-packages/glances/__init__.py", line 106, in start
    mode = GlancesMode(config=config, args=args)
  File "/usr/lib/python3.7/site-packages/glances/standalone.py", line 49, in __init__
    self.stats = GlancesStats(config=config, args=args)
  File "/usr/lib/python3.7/site-packages/glances/stats.py", line 47, in __init__
    self.load_modules(self.args)
  File "/usr/lib/python3.7/site-packages/glances/stats.py", line 101, in load_modules
    self.load_exports(args=args)
  File "/usr/lib/python3.7/site-packages/glances/stats.py", line 179, in load_exports
    config=self.config)
  File "/usr/lib/python3.7/site-packages/glances/exports/glances_influxdb.py", line 60, in __init__
    self.client = self.init()
  File "/usr/lib/python3.7/site-packages/glances/exports/glances_influxdb.py", line 70, in init
    ssl=(self.protocol.lower() == 'https'),
AttributeError: 'NoneType' object has no attribute 'lower'

This is Glances version 0.5.1 on Hassio version 0.96.4. I am writing to databases via other add-ons as well, those are not broken.

## Steps to reproduce

> Start Hassio, start Glances

## Proposed changes

> (If you have a proposed change, workaround or fix,
> describe the rationale behind it)

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/openresty/luajit2">openresty/luajit2</a></strong></div><div>OpenResty's Branch of LuaJIT 2. Contribute to openresty/luajit2 development by creating an account on GitHub.</div></blockquote>
<blockquote><div><strong><a href="https://openresty.org/en/download.html">OpenResty - Download</a></strong></div></blockquote>

### PR DESCRIPTION
**Pull request by sinclairpaul**
# Proposed Changes

Add option for specifying Influx DB protocol (`http` `https`)

Also resolves issue experienced in add on startup due to upstream issue.

## Related Issues

#25

### CODE REVIEW

**Comment by sinclairpaul** [comment]
Upstream PR https://github.com/nicolargo/glances/pull/1516

File: README.md [hunk 1]
```diff
@@ -70,7 +70,8 @@ Example add-on configuration:
     "password": "!secret glances_influxdb_password",
     "database": "glances",
     "prefix": "localhost",
-    "interval": 60
+    "interval": 60,
+    "protocol": "http"
```
Comment (frenck) [inline README.md]: Shouldn't we make this a simple `ssl` boolean to make it more consistent across the other add-ons?

Comment (sinclairpaul) [reply README.md]: It was my original plan, however ran into jq errors with it being nested.  I'll take another look today.

Comment (sinclairpaul) [reply README.md]: Not sure what challenge I was having before, but changed to boolean and made optional.

File: glances/rootfs/etc/cont-init.d/glances.sh [hunk 2]
```diff
@@ -18,6 +19,11 @@ fi
 
 # Export Glances data to InfluxDB
 if bashio::config.true 'influxdb.enabled'; then
+    if bashio::config.true 'influxdb.ssl'; then
```
Comment (frenck) [inline glances/rootfs/etc/cont-init.d/glances.sh]: An `else` can be avoided in a lot of cases:

```bash
protocol='http'
if bashio::config.true 'influxdb.ssl'; then
    protocol='https'
fi
```

**Pull request merged**